apidef/migration_test.go

```diff
@@ -205,3 +205,48 @@ func TestAPIDefinition_MigrateVersioning_StripPath(t *testing.T) {
 	})
 
 }
+
+func TestAPIDefinition_MigrateCache(t *testing.T) {
+	name := "Default"
+
+	versionInfo := VersionInfo{
+		Name:             name,
+		UseExtendedPaths: true,
+		ExtendedPaths: ExtendedPathsSet{
+			Cached: []string{"test"},
+		},
+	}
+
+	old := func() APIDefinition {
+		return APIDefinition{
+			VersionDefinition: VersionDefinition{
+				Location: URLLocation,
+			},
+			VersionData: VersionData{
+				NotVersioned:   false,
+				DefaultVersion: name,
+				Versions: map[string]VersionInfo{
+					name: versionInfo,
+				},
+			},
+		}
+	}
+
+	check := func(t *testing.T, base APIDefinition) {
+		err := base.MigrateCachePlugin()
+		assert.NoError(t, err)
+
+		defaultVersionInfo := base.VersionData.Versions[base.VersionData.DefaultVersion]
+		path1 := defaultVersionInfo.ExtendedPaths.AdvanceCacheConfig[0]
+		path1Cache := defaultVersionInfo.ExtendedPaths.Cached[0]
```

Comment:
Not just one field check. Instead, we should write an `expectedAdvCache` and compare it with actual.